src/coreclr/src/tools/Common/Compiler/SingleMethodRootProvider.cs

```diff
@@ -11,16 +12,17 @@ namespace ILCompiler
     /// </summary>
     public class SingleMethodRootProvider : ICompilationRootProvider
     {
-        private MethodDesc _method;
+        private IEnumerable<MethodDesc> _methods;
 
-        public SingleMethodRootProvider(MethodDesc method)
+        public SingleMethodRootProvider(IEnumerable<MethodDesc> methods)
```

Comment:
Can you come up with a better fitting name for this? Pretty much all the occurrences of `SingleMethod` in the entire diff are now confusing.

Reply:
@MichalStrehovsky, do you like the name "SpecifiedMethods" 

@dotnet/crossgen-contrib I would also change all of the command line option names to be specifiedmethods…

For instance --specifiedmethodstypename, --specifiedmethodsname --specifiedmethodsgenericarg

Reply:
That name works.

I was thinking about the option in general - is this actually better than addressing the TODO to add a command line option to specify an overload? Typically when I use the SingleMethod mode, I set up breakpoints on JitInterface - if this mode compiles multiple methods, it's just going to be noise we need to step through because we probably still only care about that one method. It's more convenient to set up than specifying an overload, but that's where the convenience ends.

I'm fine if you say this workflow works for you though.

Reply:
Well, this workflow works for me, and overload specification gets messy as C# starts to add things that have custom modifiers and such, especially if there are a bunch of arguments, it gets old very, very fast whereas looking at 5 methods isn't so bad. In general, I would say exact specification is more important when one wants to use a debugger, and the inexact model is more useful when doing debugging by use of r2rdump. At the moment, I'm primarily using R2RDump, so the extra methods aren't a problem. I imagine this will change over time, and I'll come back and add the ability to specify an method by overload.... and then at some point I'll add a command line switch to print out the command line that is needed to trigger compilation of a specific method....  but... one problem at a time.

Reply:
Right - it could be as simple as adding a feature that:

* If the name is ambiguous, prints all matching methods with their sigs as a numbered list (in metadata order)
* Add a command line option that lets you specify the ordinal number

It's probably also a smaller change than this, line-count wise. But I know you already wrote this version of the code, so...